Doc/tools/static/switchers.js

```diff
@@ -0,0 +1,123 @@
+      if (language == current_language)
+        buf.push('<option value="' + language + '" selected="selected">' +
+                 all_languages[current_language] + '</option>');
+      else
+        buf.push('<option value="' + language + '">' + title + '</option>');
+    });
+    buf.push('</select>');
+    return buf.join('');
+  }
+
+  function naviagate_if_exists(url, default_url) {
+    // check beforehand if url exists, else redirect to default_url.
+    $.ajax({
+      url: url,
+      success: function() {
+         window.location.href = url;
+      },
+      error: function() {
+         window.location.href = default_url;
+      }
+    });
+  }
+
+  function on_version_switch() {
+    var selected_version = $(this).children('option:selected').attr('value') + '/';
+    var url = window.location.href;
+    var current_language = find_language_in_url(url);
+    var current_version = find_version_in_url(url);
+    var new_url = url.replace('.org/' + current_language + current_version,
+                              '.org/' + current_language + selected_version);
+    if (new_url != url) {
+      naviagate_if_exists(new_url, 'https://docs.python.org/' +
+                          current_language + selected_version);
+    }
+  }
+
+  function on_language_switch() {
+    var selected_language = $(this).children('option:selected').attr('value') + '/';
+    var url = window.location.href;
+    var current_language = find_language_in_url(url);
+    var current_version = find_version_in_url(url);
+    if (selected_language == 'en/') // Special 'default' case for english.
+      selected_language = '';
+    var new_url = url.replace('.org/' + current_language + current_version,
+                              '.org/' + selected_language + current_version);
+    if (new_url != url) {
+      naviagate_if_exists(new_url, 'https://docs.python.org/' +
+                          selected_language + current_version);
+    }
+  }
+
+  // Returns the path segment as a string, like 'fr/' or '' if not found.
+  function find_language_in_url(url) {
+    var language_regexp = '\.org/(' + Object.keys(all_languages).join('|') + '/)';
+    var match = url.match(language_regexp);
+    if (match !== null)
+        return match[1];
+    return '';
+  }
+
+  // Returns the path segment as a string, like '3.6/' or '' if not found.
+  function find_version_in_url(url) {
+    var language_segment = '(?:(?:' + Object.keys(all_languages).join('|') + ')/)';
+    var version_segment = '(?:(?:\\d|py3k|dev|(?:(?:release/)?\\d\\.\\d[\\w\\d\\.]*))/)';
+    var version_regexp = '\\.org/' + language_segment + '?(' + version_segment + ')';
+    var match = url.match(version_regexp);
+    if (match !== null)
+      return match[1];
+    return ''
+  }
+
+  $(document).ready(function() {
+    var release = DOCUMENTATION_OPTIONS.VERSION;
+    var current_language = find_language_in_url(window.location.href).replace(
+            /\/+$/g, '') || 'en';
```

Comment:
I disambiguated it by renaming functions and creating an intermediate named variable.

`find_language_in_url`, now called `language_segment_from_url` extract a path segment like `3.6/`, not a language tag, so we have to strip the `/` to get the language tag.

This is usefull to disambiguate between having no language in URL being the empty string versus having a language in the URL being "LANGUAGE_TAG/", so we can later concatenate "language + version" and it gives the expected result:

- '' + "3.5/" → "3.5/"
- "fr/" + "" → "fr/"
- "fr/" + "3.5/" → "fr/3.5/"

this permit to easily find and replace an optional language and an optional version in an URL like:
```
var new_url = url.replace('.org/' + current_language + current_version,
                          '.org/' + selected_language + current_version);
```